packages/intrepid2/src/Projection/Intrepid2_ProjectionTools.hpp

```diff
@@ -780,15 +780,27 @@ class ProjectionTools {
       Teuchos::LAPACK<ordinal_type,valueType> lapack_;
       ordinal_type numCells = basisCoeffs.extent(0);
 
+      // due to the particular type of elemDof (possibly due either to layout or subviews having been
+      // taken), the usual pattern of create_mirror_view_and_copy() does not work "out of the box".
+      // So we do a manual copy on device, followed by the create_mirror_view_and_copy().
+      Kokkos::View<ordinal_type*> elemDofCopy("elemDofCopy", elemDof.extent(0));
```

Comment:
We do not allocate a view with default space. All Intrepid2 is templated with device type and the view allocation should respect the device type.